src/NUnitFramework/framework/Constraints/RangeConstraint.cs

```diff
@@ -33,26 +33,41 @@ namespace NUnit.Framework.Constraints
     /// </summary>
     public class RangeConstraint : Constraint
     {
-        private readonly IComparable from;
-        private readonly IComparable to;
+        private readonly object from;
+        private readonly object to;
 
         private ComparisonAdapter comparer = ComparisonAdapter.Default;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="RangeConstraint"/> class.
         /// </summary>
-        /// <remarks>from must be less than or equal to true</remarks> 
-        /// <param name="from">Inclusive beginning of the range. Must be less than or equal to to.</param>
-        /// <param name="to">Inclusive end of the range. Must be greater than or equal to from.</param>
+        /// <remarks>The <paramref name="from"/> value must be less than or equal to the <paramref name="to"/> value.</remarks> 
+        /// <param name="from">Must be less than or equal to the <paramref name="to"/> value.</param>
+        /// <param name="to">Must be greater than or equal to the <paramref name="from"/> value.</param>
         public RangeConstraint(IComparable from, IComparable to) : base( from, to )
         {
             // Issue #21 - https://github.com/nunit/nunit-framework/issues/21
             // from must be less than or equal to to
-            if (comparer.Compare(from, to) > 0)
-                throw new ArgumentException( "from must be less than to" );
 
             this.from = from;
             this.to = to;
+            CompareFromAndTo();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RangeConstraint"/> class for objects that did not implement <see cref="IComparable"/>.
+        /// </summary>
+        /// <remarks>The <paramref name="from"/> value must be less than or equal to the <paramref name="to"/> value.</remarks> 
+        /// <param name="from">Must be less than or equal to the <paramref name="to"/> value.</param>
+        /// <param name="to">Must be greater than or equal to the <paramref name="from"/> value.</param>
+        /// <param name="comparer">Class that implements <seealso cref="IComparer"/> used to compare the objects.</param>
+        public RangeConstraint(object from, object to, IComparer comparer) : base(from, to)
+        {
```

Comment:
Actually, it would make a lot of sense if the normal way to use it were through the constructor. But for us the constructors are there but not primary. Construction is generally through the fluent syntax.